exercises/allergies/.gitignore

```diff
@@ -0,0 +1,7 @@
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
```

Comment:
i prefer to stick to convention over something custom here, but i'd be happy with what is already in the PR or the custom msg you described, leaning slightly towards what's already here